### PR TITLE
Improve loading of third party editor control CEMS-2181

### PR DESCRIPTION
**Pull request by steve-giles**
This is a technical change which improves how the Angular Custom Element accesses the third party CKEditor Control.
1. CKEditor code is now accessed locally from "assets/ckeditor" instead of from a CDN.
2. Unit tests have been fixed. 
3. Editor config has been moved from the component to /assets/ckeditor/config.js.
4. Drop and drag plugin has been moved from the component to /assets/ckeditor/plugins/hcard.

There are two files to review for the component which includes changes to fix the unit tests and to move editor configuration settings out of the component and into the "assets/ckeditor" directory where the editor is located.  Also fixes an issue where the editor wasn't getting destroyed.
- src/app/dts/template-editor/template-editor.component.spec.ts 
- src/app/dts/template-editor/template-editor.component.ts

Specifices the location of the CKEditor source code
- src/ckeditor.js

Specifies the configuration of our editor and adds the drag and drop plugin (hcard)
- src/assets/ckeditor/config.js

Contains code for our drag and drop plugin.  This is the directory where we can place custom plugins.  It's much better to place them here than in the component.  In a future PR I'm going to rename "hcard" to something more relevant.
- src/assets/ckeditor/plugins/hcard/plugin.js

Tells Karama (unit tests) where the CKEditor source code is located so it can create the editor.
- karma.conf.json

Supporting files 
- angular.json
- build-script.json

For testing to demonstrate how the custom element is referenced (i.e. from CEMS)
- src/index.html

### CODE REVIEW

File: debug.log [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[1223/133143.352:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
+[1223/135737.186:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
+[1223/141917.691:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
```
Comment (alcfeoh) [inline debug.log]: This file should most liikely not be checked in

**Review by alcfeoh — COMMENTED** [review]
Most of that code seems to be the CK editor checked in into our code. Shouldn't it be an NPM dependency instead? Which files should I actually review out of those 758?

File: debug.log (hunk 1 above)
Comment (steve-giles) [reply debug.log]: Fixed.

**Comment by steve-giles** [comment]
> Most of that code seems to be the CK editor checked in into our code. Shouldn't it be an NPM dependency instead? Which files should I actually review out of those 758?

There are two files to review for the component which includes changes to fix the unit tests and to move editor configuration settings out of the component and into the "assets/ckeditor" directory where the editor is located.  Also fixes an issue where the editor wasn't getting destroyed.
- src/app/dts/template-editor/template-editor.component.spec.ts 
- src/app/dts/template-editor/template-editor.component.ts

Specifices the location of the CKEditor source code
- src/ckeditor.js

Specifies the configuration of our editor and adds the drag and drop plugin (hcard)
- src/assets/ckeditor/config.js

Contains code for our drag and drop plugin.  This is the directory where we can place custom plugins.  It's much better to place them here than in the component.  In a future PR I'm going to rename "hcard" to something more relevant.
- src/assets/ckeditor/plugins/hcard/plugin.js

Tells Karama (unit tests) where the CKEditor source code is located so it can create the editor.
- karma.conf.json

Supporting files 
- angular.json
- build-script.json

For testing to demonstrate how the custom element is referenced (i.e. from CEMS)
- src/index.html

**Review by alcfeoh — APPROVED** [review]

**Pull request merged**